application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,160 @@
+                    "do you wish for your message to stay private", false);
+
+        channelIdToLastCommandInvocation = setCooldown();
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    @NotNull
+    private Cache<Long, Instant> setCooldown() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandInteractionEvent event) {
+        MessageChannel messageChannel = event.getChannel();
+        if (isChannelOnCooldown(messageChannel)) {
+            event
+                .reply("Please wait a bit, this command can only be used once per %d %s.".formatted(
+                        COOLDOWN_DURATION_VALUE,
+                        COOLDOWN_DURATION_UNIT.toString().toLowerCase(Locale.US)))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+        channelIdToLastCommandInvocation.put(messageChannel.getIdLong(), Instant.now());
+
+        Optional<TextChannel> auditLogChannel = getChannel(event);
+        if (auditLogChannel.isEmpty()) {
+            return;
+        }
+
+        List<ModAuditLogWriter.Attachment> attachments = getAttachments(event);
+
+        String user = getName(event);
+        MessageAction message = auditLogChannel.orElseThrow().sendMessageEmbeds(messageEmbed(user));
+
+        message = buildAttachment(attachments, message);
+        message.queue();
+
+        event.reply("Thank you for contacting the moderators").setEphemeral(true).queue();
+
+    }
+
+    @NotNull
+    private String getName(@NotNull SlashCommandInteractionEvent event) {
+        String user = "@" + event.getUser().getName();
+        boolean optionalMessage =
+                Objects.requireNonNull(event.getOption(OPTION_MESSAGE_PRIVATE)).getAsBoolean();
+        if (optionalMessage) {
+            user = "Anonymous";
+        }
+        return user;
+    }
+
+    private MessageAction buildAttachment(List<ModAuditLogWriter.Attachment> attachments,
+            MessageAction message) {
+        for (ModAuditLogWriter.Attachment attachment : attachments) {
+            message = message.addFile(attachment.getContentRaw(), attachment.name());
+        }
+        return message;
+    }
+
+    @NotNull
+    private MessageEmbed messageEmbed(String user) {
+        return new EmbedBuilder().setAuthor("Modmail Command invoked")
+            .setColor(Color.BLACK)
+            .setTitle("/modmail from '%s' ".formatted(user))
+            .build();
+    }
+
+    @NotNull
+    private List<ModAuditLogWriter.Attachment> getAttachments(
+            @NotNull SlashCommandInteractionEvent event) {
+        String content = Objects.requireNonNull(event.getOption(MESSAGE)).getAsString();
```

Comment:
Again, why the requireNonNull?

Better question, what does that method do?

It throws an empty NPE if the input is null.

So you create an empty NPE without info on *what* is null, or *where* something is null.

Why not just let Java throw the NPE then (since like Java 14 they have been improved a lot)? Or add a comment to the exceptions (Objects#requireNonNull(T, String) takes a comment, I believe you should use either of those instead.

Reply:
The method uses the message the user has sent , adds it to the attachments and retrieves it to the array 

I added it because it gets rid of the warning, and because the message can't be null since its required (right?)

Reply:
The message can't be null either way, cause of a NPE.

I'd appreciate it if you added a comment to the exception, or you removed the requireNonNull as mentioned above

Reply:
No problem. Il remove it if there is smth I should add with it lmk.

Reply:
Done